src/common/webview_chromium.cpp

```diff
@@ -0,0 +1,809 @@
+}
+
+bool wxWebViewChromium::IsBusy() const
+{
+    if(m_clientHandler->GetBrowser())
+        return m_clientHandler->GetBrowser()->IsLoading();
+    else
+        return false;
+}
+
+void wxWebViewChromium::SetEditable(bool enable)
+{
+    wxString mode = enable ? "\"on\"" : "\"off\"";
+    RunScript("document.designMode = " + mode); 
+}
+
+void wxWebViewChromium::DoSetPage(const wxString& html, const wxString& baseUrl)
+{
+    m_clientHandler->GetBrowser()->GetMainFrame()->LoadString(html.ToStdString(),
+                                                              baseUrl.ToStdString());
+}
+
+wxWebViewZoom wxWebViewChromium::GetZoom() const
+{
+     return m_zoomLevel;
+}
+
+
+void wxWebViewChromium::SetZoom(wxWebViewZoom zoom)
+{
+    m_zoomLevel = zoom;
+
+    double mapzoom;
+    // arbitrary way to map our common zoom enum to float zoom
+    switch ( zoom )
+    {
+        case wxWEBVIEW_ZOOM_TINY:
+            mapzoom = -1.0;
+            break;
+
+        case wxWEBVIEW_ZOOM_SMALL:
+            mapzoom = -0.5;
+            break;
+
+        case wxWEBVIEW_ZOOM_MEDIUM:
+            mapzoom = 0.0;
+            break;
+
+        case wxWEBVIEW_ZOOM_LARGE:
+            mapzoom = 0.5;
+            break;
+
+        case wxWEBVIEW_ZOOM_LARGEST:
+            mapzoom = 1.0;
+            break;
+
+        default:
+            wxASSERT(false);
+    }
+    m_clientHandler->GetBrowser()->GetHost()->SetZoomLevel(mapzoom);
+}
+
+void wxWebViewChromium::SetZoomType(wxWebViewZoomType type)
+{
+    // there is only one supported zoom type at the moment so this setter
+    // does nothing beyond checking sanity
+    wxASSERT(type == wxWEBVIEW_ZOOM_TYPE_LAYOUT);
+}
+
+wxWebViewZoomType wxWebViewChromium::GetZoomType() const
+{
+    return wxWEBVIEW_ZOOM_TYPE_LAYOUT;
+}
+
+bool wxWebViewChromium::CanSetZoomType(wxWebViewZoomType type) const
+{
+    return type == wxWEBVIEW_ZOOM_TYPE_LAYOUT;
+}
+
+void wxWebViewChromium::RegisterHandler(wxSharedPtr<wxWebViewHandler> handler)
+{
+    CefRegisterSchemeHandlerFactory( handler->GetName().ToStdWstring(), "",
+                                     new SchemeHandlerFactory(handler) );
+}
+
+#ifdef __WXMSW__
+bool wxWebViewChromium::StartUp(int &code, const wxString &path)
+#else
+bool wxWebViewChromium::StartUp(int &code, const wxString &path,
+                                int argc, char* argv[])
+#endif
+{
+#ifdef __WXMSW__
+    CefMainArgs args(wxGetInstance());
+#else
+    CefMainArgs args(argc, argv);
+#endif
+    // If there is no subprocess then we need to execute on this process
+    if ( path == "" )
+    {
+#if CHROME_VERSION_BUILD >= 1750
+        code = CefExecuteProcess(args, NULL, NULL);
+#else
+        code = CefExecuteProcess(args, NULL);
+#endif
+        if ( code >= 0 )
+            return false;
+    }
+
+    CefSettings settings;
+#if CHROME_VERSION_BUILD >= 1750
+    settings.no_sandbox = true;
+#endif
+
+#ifdef __WXDEBUG__
```

Comment:
This is not appropriate, `__WXDEBUG__` is defined even in release build by default in wx3.

It would probably be better to either use some `wxWEBVIEW_DEBUG` environment variable or provide a method for enabling debug logging.